src/background.js

```diff
@@ -1,5 +1,196 @@
-browser.browserAction.onClicked.addListener(async () => {
-  await browser.tabs.create({
-    url: "debug-page/index.html",
-  });
+const configByAddonId = {
+  "messaging-system-personalization-experiment-1-addon-control@mozilla.org": {
+    branch: "control",
+    asRouterCfrProviderConfig: {
+      bucket: "cfr-control",
+      cohort: "PERSONALIZATION_EXPERIMENT_1_CONTROL",
+    },
+  },
+  "messaging-system-personalization-experiment-1-addon-treatment@mozilla.org": {
+    branch: "treatment",
+    asRouterCfrProviderConfig: {
+      bucket: "cfr-experiment",
+      cohort: "PERSONALIZATION_EXPERIMENT_1_TREATMENT",
+    },
+    scoreThreshold: 1.6,
+  },
+};
+const config = configByAddonId[browser.runtime.id];
+const { bucket, cohort } = config.asRouterCfrProviderConfig;
+console.info("Experiment add-on background.js", browser.runtime.id);
+console.debug({ config });
+
+const onError = e => {
+  console.error(e);
+  browser.normandyAddonStudy.endStudy("CAUGHT_ERROR");
+};
+
+const run = async () => {
+  try {
+    // TODO - double-check that study has not already ended
+    const study = await browser.normandyAddonStudy.getStudy();
+    console.debug({ study });
+
+    const { hasRunInitiationAtLeastOnce } = await browser.storage.local.get(
+      "hasRunInitiationAtLeastOnce",
+    );
+    if (hasRunInitiationAtLeastOnce !== true) {
+      await firstRun();
+      await browser.storage.local.set({ hasRunInitiationAtLeastOnce: true });
+    }
+    await everyRun();
+  } catch (e) {
+    onError(e);
+  }
+};
+
+const firstRun = async () => {
+  try {
+    if (config.branch === "control") {
+      console.info("Writing hard-coded provider cfr pref", { bucket, cohort });
+      await browser.privileged.messagingSystem.setASRouterCfrProviderPref(
+        bucket,
+        cohort,
+      );
+      return;
+    }
+
+    console.info("Writing hard-coded score threshold", config.scoreThreshold);
+    await browser.privileged.personalizedCfrPrefs.setScoreThreshold(
+      config.scoreThreshold,
+    );
+  } catch (e) {
+    onError(e);
+  }
+};
+
+const everyRun = async () => {
+  try {
+    // Do nothing on every run in control branch
+    if (config.branch === "control") {
+      return;
+    }
+
+    // Imports models from remote settings (`cfr-models` bucket)
+    // Drives update cycle so that the computations gets updated on each update
+    console.info("Subscribing to model updates from remote settings");
+    browser.privileged.messagingSystem.onCfrModelsSync.addListener(
+      onCfrModelsSync,
+    );
+
+    // Force sync here on add-on startup / browser startup?
```

Comment:
For every run, I think it's reasonable to rely on our regular update cycle. However, we may need to rely on using polling for regular updates (e.g. every hour) due to technical limitations in the RS infra.